yew-macro/src/html_classes/mod.rs

```diff
@@ -0,0 +1,34 @@
+use proc_macro2::TokenStream;
+use quote::{quote, ToTokens};
+use syn::parse::{Parse, ParseStream, Result};
+use syn::punctuated::Punctuated;
+use syn::token::Comma;
+use syn::Expr;
+
+/// List of HTML classes.
+pub struct HtmlClasses(Punctuated<Expr, Comma>);
+
+impl Parse for HtmlClasses {
+    fn parse(input: ParseStream) -> Result<Self> {
+        Ok(HtmlClasses(Punctuated::<Expr, Comma>::parse_terminated(
+            input,
+        )?))
+    }
+}
+
+impl ToTokens for HtmlClasses {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let n = self.0.len();
+        let classes = self.0.iter();
+        let new_tokens = quote! {
+            let mut __yew_classes = ::yew::virtual_dom::Classes::with_capacity(#n);
+            #(__yew_classes.push(#classes);)*
+            __yew_classes
+        };
+
+        tokens.extend(quote! {{
+            #[allow(clippy::useless_conversion, unused_braces)]
```

Comment:
Fixed in af582f4

Reply:
I'm not sure about this.

```rust
    fn to_tokens(&self, tokens: &mut TokenStream) {
        let n = self.0.len();
        let classes = self.0.iter();
        let new_tokens = quote! {
            let mut __yew_classes = ::yew::virtual_dom::Classes::with_capacity(#n);
            #(__yew_classes.push(#classes);)*
            __yew_classes
        };

        tokens.extend(quote! {{
            #new_tokens
        }});
    }
```

I think if classes is empty, it should complain that `__yew_classes` doesn't need to be mutable. But I couldn't trigger the warning.

Reply:
It probably does emit that warning but Clippy is actually pretty smart, it ignores the error if it doesn't originate from user code tokens. More specifically, because you're using `quote!` here all the tokens (apart from `classes`) default to the call site span which Clippy knows to ignore.
Sometimes this doesn't work because we're using `quote_spanned!` which attaches a span to all the tokens (which is, of course, most often the span from an input token which makes Clippy think that the user is at fault).

Reply:
Good to know... thanks!